remove map bounds

sevko opened the pull request:
Remove all of the configuration options that bounded the scrollable map area, since they were put in place to prevent out-of-bounds longitude parameters from triggering API parameter errors and that's no longer an issue with pelias/api#121 .

fixes pelias/api#56

sevko committed:
Remove scroll bounds.

js/demo.js
	-Remove all of the configuration that bounded the Leaflet
	mapping, preventing it from mapping/users from scrolling outside
	of the typical lat/lng bounds.

sevko committed:
Emit 'map.setView' after reverse-geocoding a point.

js/demo.js
	-The absence of a `map.setView` event was causing the screen to
	awkwardly center on the middle of the world map while a marker
	appeared elsewhere when requests from outside the normal
	longitude bounds were made.

missinglink commented:
pulling this `in review` as it's a PR that's been open for some time. please check it is still relevant considering subsequent changes in both api/demo before merging.

@hkrishna may know more about it?

orangejulius commented:
Yeah, this code is correct, assuming the API supports wrapping lat/lon. If the API doesn't, the demo will have weird behavior where searches won't return any results unless you pan left/right the correct amount to make sure the centerpoint is on the "real" lat/lon.

orangejulius committed:
Enable worldCopyJump Leaflet setting

It's not perfect, but this allows markers to be seen on the map
regardless of how many times the user has panned around the earth. There
is a little bit of lag/graphical tearing but not too bad. Better than
markers not showing up for sure.

orangejulius commented:
I added one commit to this PR that causes Leaflet to automatically reposition the map such that if you are rotating past the bounds of the map, it will seamlessly place the center of the map back onto the same location, but with a longitude between -180 and 180. There are some slight graphical tearing issues when this happens but they aren't really that bad. The alternative is to have a map that can scroll infinitely, but behaves strangely:

* When you search for something, results won't show up because they are on "another copy" of the earth.
* Clicking on a search result moves you from one "copy" of the earth to another.

worldCopyJump seems to mostly fix these.

dianashk commented:
:+1:
